test-framework/junit5-component/src/main/java/io/quarkus/test/component/QuarkusComponentTestConfiguration.java

```diff
@@ -74,23 +78,36 @@ QuarkusComponentTestConfiguration update(Class<?> testClass) {
                 }
             }
         }
-        // All fields annotated with @Inject represent component classes
         Class<?> current = testClass;
-        while (current != null) {
+        while (current != null && current != Object.class) {
+            // All fields annotated with @Inject represent component classes
             for (Field field : current.getDeclaredFields()) {
                 if (field.isAnnotationPresent(Inject.class) && !resolvesToBuiltinBean(field.getType())) {
                     componentClasses.add(field.getType());
                 }
             }
-            current = current.getSuperclass();
-        }
-        // All static nested classes declared on the test class are components
-        if (addNestedClassesAsComponents) {
-            for (Class<?> declaredClass : testClass.getDeclaredClasses()) {
-                if (Modifier.isStatic(declaredClass.getModifiers())) {
-                    componentClasses.add(declaredClass);
+            // All static nested classes declared on the test class are components
+            if (addNestedClassesAsComponents) {
+                for (Class<?> declaredClass : current.getDeclaredClasses()) {
+                    if (Modifier.isStatic(declaredClass.getModifiers())) {
+                        componentClasses.add(declaredClass);
+                    }
                 }
             }
+            // All params of test methods but not TestInfo, not annotated with @InjectMock and not annotated with @SkipInject
+            for (Method method : current.getDeclaredMethods()) {
+                if (method.isAnnotationPresent(Test.class)) {
+                    for (Parameter param : method.getParameters()) {
+                        if (param.getType() == TestInfo.class
```

Comment:
There's more built-in JUnit parameters and we might want to filter all of them here.
It should be these types - `TestInfo`, [`RepetitionInfo`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/RepetitionInfo.html) and [`TestReporter`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/TestReporter.html).
Plus you can also have a parameter annotated with [`@TempDir`](https://junit.org/junit5/docs/current/user-guide/#writing-tests-built-in-extensions-TempDirectory) which is something JUnit handles as well.